crates/prover/src/runner.rs

```diff
@@ -432,13 +436,15 @@ where
             da_data.iter_mut().for_each(|blob| {
                 blob.full_data();
             });
-            let sequencer_commitments: Vec<SequencerCommitment> =
+            let mut sequencer_commitments: Vec<SequencerCommitment> =
                 self.extract_sequencer_commitments(l1_block.header().hash().into(), &mut da_data);
+            // Make sure all sequencer commitments are stored in ascending order.
+            sequencer_commitments.sort_by_key(|commitment| commitment.l2_start_block_number);
```

Comment:
good catch, however if commitments in a group are not in same order also in da, it will break the proving. I'll make an issue about it 👍🏻